src/core/hw/aes/key.cpp

```diff
@@ -65,17 +89,46 @@ struct KeySlot {
 std::array<KeySlot, KeySlotID::MaxKeySlotID> key_slots;
 std::array<std::optional<AESKey>, 6> common_key_y_slots;
 
-AESKey HexToKey(const std::string& hex) {
-    if (hex.size() < 32) {
-        throw std::invalid_argument("hex string is too short");
-    }
+enum class FirmwareType : u32 {
+    ARM9 = 0,  // uses NDMA
+    ARM11 = 1, // uses XDMA
+};
 
-    AESKey key;
-    for (std::size_t i = 0; i < key.size(); ++i) {
-        key[i] = static_cast<u8>(std::stoi(hex.substr(i * 2, 2), 0, 16));
-    }
+struct FirmwareSectionHeader {
+    u32_le offset;
+    u32_le phys_address;
+    u32_le size;
+    enum_le<FirmwareType> firmware_type;
+    std::array<u8, 0x20> hash; // SHA-256 hash
+};
 
-    return key;
+struct FIRM_Header {
+    u32_le magic;         // FIRM
+    u32_le boot_priority; // Usually 0
+    u32_le arm11_entrypoint;
+    u32_le arm9_entrypoint;
+    INSERT_PADDING_BYTES(0x30);                           // Reserved
+    std::array<FirmwareSectionHeader, 4> section_headers; // 1st ARM11?, 3rd ARM9
+    std::array<u8, 0x100> signature; // RSA-2048 signature of the FIRM header's hash
+};
+
+struct ARM9_HEADER {
```

Comment:
```suggestion
struct ARM9_Header {
```
for consistent style with `FIRM_Header`